### PR TITLE
{bio} [intel/2017a] Trinity 2.5.1

### PR DESCRIPTION
**Pull request by smoors**

### CODE REVIEW

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name = 'Trinity'
+version = '2.5.1'
+
+homepage = 'http://trinityrnaseq.github.io'
+description = """Trinity represents a novel method for the efficient and robust de novo reconstruction
+ of transcriptomes from RNA-Seq data. Trinity combines three independent software modules: Inchworm,
+ Chrysalis, and Butterfly, applied sequentially to process large volumes of RNA-Seq reads."""
+toolchain = {'name': 'intel', 'version': '2017a'}
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: please include an empty line above

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+toolchainopts = {'optarch': True}
+preinstallopts = "JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8 "
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: This should be moved down, below the `dependencies` block (with an empty line above).

Maybe also include a comment why this is needed?

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+source_urls = ['https://github.com/trinityrnaseq/trinityrnaseq/archive/']
+sources = ['%(name)s-v%(version)s.tar.gz']
+patches = [
+    'chrysalis_commandline_noconsts_2012-10-05.patch',
+    'Trinity-2.5.1_plugins-Makefile.patch',
+    'Trinity-2.5.1_icpc.patch',
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: please avoid hardcoding `2.5.1` here, use `%(version)s` instead

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+checksums = [
+    'bf14f96981bb028d7960dec895a1bcbda1fadbbe562a913b188559fd74b90457',  # Trinity-v2.5.1.tar.gz
+    # chrysalis_commandline_noconsts_2012-10-05.patch
+    '3442803c357576557de843c80b73436e59e9a99027d60f5f63bb3368eceb4a5c',
+    '2682f182264bf101d0a2b182ca4110f2d1ab16c56e69e804379023c1e1a48847',  # Trinity-2.5.1_plugins-Makefile.patch
+    'a686eafb11d063b8a604190389db98808aaedf3458973b4124c38c9019878901',  # Trinity-2.5.1_icpc.patch
+]
+
+builddependencies = [
+    ('Autotools', '20150215', '', ('GCCcore', '6.3.0')),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: there should be no need to hardcode the toolchain, just use `('Autotools', '20150215')`?

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+]
+
+dependencies = [
+    ('Java', '1.8.0_121', '', True),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: use latest `1.8.0_152`?

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb (hunk 2 above)
Comment (smoors) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: This option did not work, I forgot to remove it. I had to set it in the easyblock file, see
https://github.com/easybuilders/easybuild-easyblocks/pull/1302

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb (hunk 5 above)
Comment (smoors) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: done

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb (hunk 4 above)
Comment (smoors) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: done

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb (hunk 3 above)
Comment (smoors) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: done

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb (hunk 1 above)
Comment (smoors) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: done

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb (hunk 5 above)
Comment (smoors) [reply easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: The latest ant-compatible version of Java is 1.8.0_144

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+name = 'Trinity'
+version = '2.5.1'
+
+homepage = 'http://trinityrnaseq.github.io'
+description = """Trinity represents a novel method for the efficient and robust de novo reconstruction
+ of transcriptomes from RNA-Seq data. Trinity combines three independent software modules: Inchworm,
+ Chrysalis, and Butterfly, applied sequentially to process large volumes of RNA-Seq reads."""
+
+toolchain = {'name': 'intel', 'version': '2017a'}
+toolchainopts = {'optarch': True}
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: @smoors Please remove this line, `optarch` is already enabled by default.

File: easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    '2682f182264bf101d0a2b182ca4110f2d1ab16c56e69e804379023c1e1a48847',  # Trinity-2.5.1_plugins-Makefile.patch
+    'a686eafb11d063b8a604190389db98808aaedf3458973b4124c38c9019878901',  # Trinity-2.5.1_icpc.patch
+]
+
+builddependencies = [
+    ('Autotools', '20150215'),
+]
+
+dependencies = [
+    ('Java', '1.8.0_144', '', True),
```
Comment (boegel) [inline easybuild/easyconfigs/t/Trinity/Trinity-2.5.1-intel-2017a.eb]: Use latest `1.8.0_152`?

You will need to include an `ant` easyconfig that uses it (make that latest `ant` version too).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2410.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1f8727175de526020d6e8ba8fe7389ac for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2085.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/d77a4311422e7a61b8b0a72f4146b327 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Pull request merged**